app/terminal/terminal.tsx

```diff
@@ -72,7 +80,27 @@ export default class TerminalComponent extends React.Component<TerminalProps, Te
   }
 
   handleDownloadClicked() {
-    var element = document.createElement("a");
+    if (this.props.fullLogsFetcher) {
+      this.setState({ isLoadingFullLog: true });
+      const element = document.createElement("a");
+      this.props
+        .fullLogsFetcher()
+        .then((fullLog: string) => {
+          const unstyledLogs = fullLog.replace(ANSI_STYLES_REGEX, "");
+          element.setAttribute("href", "data:text/plain;charset=utf-8," + encodeURIComponent(unstyledLogs));
```

Comment:
Seems like this is all pretty much the same code as below - should we pull it into a function?

Reply:
Done.